ksqldb-engine/src/test/java/io/confluent/ksql/engine/generic/GenericRecordFactoryTest.java

```diff
@@ -43,21 +43,35 @@
 import java.util.List;
 import java.util.concurrent.atomic.AtomicLong;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.MockitoRule;
 
+@RunWith(MockitoJUnitRunner.class)
 public class GenericRecordFactoryTest {
 
+  @Rule public MockitoRule rule = MockitoJUnit.rule();
+
   private static final ColumnName KEY = ColumnName.of("K0");
   private static final ColumnName COL0 = ColumnName.of("COL0");
   private static final ColumnName COL1 = ColumnName.of("COL1");
 
   private final AtomicLong clock = new AtomicLong();
-  private final KsqlConfig config = new KsqlConfig(ImmutableMap.of());
   private final FunctionRegistry functions = TestFunctionRegistry.INSTANCE.get();
-  private final GenericRecordFactory recordFactory = new GenericRecordFactory(config, functions, clock::get);
+
+  @Mock
+  private KsqlConfig ksqlConfig;
+
+  //fixme: why is config not mocked yet here?
+  private GenericRecordFactory recordFactory;
 
   @Before
   public void setUp() {
+    recordFactory = new GenericRecordFactory(ksqlConfig, functions, clock::get);
```

Comment:
It's customary to have this be the last line in `setUp()`, since this is initializing the variable being tested (so typically all the other setup should happen first).